lib/costellation.dart

```diff
@@ -0,0 +1,642 @@
+    G_FFIResult_Null result =
+        bindings.item_rename(_pointer, name.toNativeUtf8().cast<Char>());
+
+    if (result.error != nullptr) {
+      throw WarpException(result.error);
+    }
+  }
+
+  void setItemDescription(String description) {
+    bool result = bindings.item_set_description(
+        _pointer, description.toNativeUtf8().cast<Char>());
+
+    if (result == 0) {
+      throw Exception("Setting of item is failed");
+    }
+  }
+
+  void setItemSize(int size) {
+    G_FFIResult_Null result = bindings.item_set_size(_pointer, size);
+
+    if (result.error != nullptr) {
+      throw WarpException(result.error);
+    }
+  }
+
+  Directory itemToDirectory() {
+    G_FFIResult_Directory result = bindings.item_into_directory(_pointer);
+    if (result.error != nullptr) {
+      throw WarpException(result.error);
+    }
+    Directory directory = Directory(result.data);
+
+    return directory;
+  }
+
+  File itemToFile(Item item) {
+    G_FFIResult_File result = bindings.item_into_file(item.pointer());
+
+    if (result.error != nullptr) {
+      throw WarpException(result.error);
+    }
+    File file = File(result.data);
+
+    return file;
+  }
+
+  bool itemIsDirectory() {
+    bool result = bindings.item_is_directory(_pointer);
+
+    if (result == 0) {
+      return false;
+    }
+
+    return true;
+  }
+
+  bool itemIsFile() {
+    bool result = bindings.item_is_file(_pointer);
+
+    if (result == 0) {
+      return false;
+    }
+
+    return true;
+  }
+
+  Pointer<G_Item> pointer() {
+    return _pointer;
+  }
+
+  void drop() {
+    bindings.item_free(_pointer);
+  }
+}
+
+class File {
+  late String id;
+  late String name;
+  late int size;
+  late String description;
+  late DateTime creation;
+  late DateTime modified;
+  late String hash;
+  late String? reference;
+  late Pointer<G_File> _pointer;
+
+  File(this._pointer) {
+    Pointer<G_Item> item = bindings.file_into_item(_pointer);
+    Pointer<Char> pId = bindings.item_id(item);
+    id = pId.cast<Utf8>().toDartString();
+    Pointer<Char> pName = bindings.item_name(item);
+    name = pName.cast<Utf8>().toDartString();
+    size = bindings.item_size(item);
+    Pointer<Char> pDescription = bindings.item_description(item);
+    description = pDescription.cast<Utf8>().toDartString();
+    Pointer<Char> pCreation = bindings.item_creation(item);
+    String creationString = pCreation.cast<Utf8>().toDartString();
+    creation =
+        DateTime.parse(creationString.substring(0, creationString.length - 4));
+    Pointer<Char> pModified = bindings.item_modified(item);
+    String modifiedString = pModified.cast<Utf8>().toDartString();
+    modified =
+        DateTime.parse(modifiedString.substring(0, modifiedString.length - 4));
+    hash = item.hashCode.toString();
+
+    calloc.free(pId);
+    calloc.free(pName);
+    calloc.free(pDescription);
+    calloc.free(pCreation);
+    calloc.free(pModified);
+  }
+
+  File setDescription(String descritpion) {
```

Comment:
Not sure I understand but could you explain why youre returning `File`?

Reply:
Now I simply replace the pointer already stored and added a function to get the most update file pointer